source/iterator/cdc_iterator.go

```diff
@@ -0,0 +1,137 @@
+/*
+Copyright © 2022 Meroxa, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package iterator
+
+import (
+	"context"
+	"errors"
+	"time"
+
+	sdk "github.com/conduitio/conduit-connector-sdk"
+	"github.com/conduitio/conduit-connector-zendesk/source/position"
+	"github.com/conduitio/conduit-connector-zendesk/zendesk"
+	"gopkg.in/tomb.v2"
+)
+
+type ZendeskCursor interface {
+	FetchRecords(ctx context.Context) ([]sdk.Record, error)
+}
+
+//go:generate mockery --name=ZendeskCursor
+
+type CDCIterator struct {
+	lastModifiedTime time.Time         // ticket last updated time
+	tomb             *tomb.Tomb        // new tomb
+	ticker           *time.Ticker      // records time interval for next iteration
+	caches           chan []sdk.Record // cache to store array of tickets
+	buffer           chan sdk.Record   // buffer to store individual ticket object
```

Comment:
In another PR we had a similar thing, so I guess the reason is the same: we get a slice of records from ZenDesk, put that into caches, and then put one by one into buffer, and then send to Conduit?

If so, then why not take the slice of records from ZenDesk, iterate over them and put one by one into buffer directly?

Reply:
We can maybe take it one step further, and eliminate channels altogether. We can have a buffer, which is simply a slice. Next() could read from the slice. If there's something in it, it returns it, and removes from the buffer. If the buffer is empty, it tries to fetch more records from ZenDesk.

Reply:
For the purpose of having 2nd cache of records ready when the first cache of records is done processing by the conduit, we use a channel cache of records array.
Reference: https://github.com/ConduitIO/conduit-connector-s3/blob/main/source/iterator/cdc_iterator.go#L36-L39
Comment from maha: https://github.com/gopherslab/conduit-connector-redis/pull/8/files#diff-e24a48fcc6c1fe4841b2397219b35a1f4205ffd25060a0d2189be5e5a5077b77R147

Reply:
@maha-hajja what are your thoughts here?